File: 88-merge-sorted-array/merge-sorted-array.ts
function merge(nums1: number[], m: number, nums2: number[], n: number): void {
  let p1 = m - 1;
  let p2 = n - 1;
  let i = nums1.length - 1;
  while(p1 >= 0 && p2 >= 0) {
    if(nums2[p2] >= nums1[p1]) nums1[i --] = nums2[p2 --];
    else nums1[i --] = nums1[p1 --];
  }
  if(p1 < 0 && p2 >= 0) while(p2 >= 0) nums1[i --] = nums2[p2 --];
};

File: 88-merge-sorted-array/merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int l = nums1.length - 1;
        m = m -1;
        n = n - 1;
        while(l >= 0) {
            if(m < 0 || n < 0) break;
            if(nums1[m] < nums2[n]) {
                nums1[l --] = nums2[n --];
                continue;
            }
            else {
                nums1[l --] = nums1[m --];
                continue;
            }
        }
        while(m >= 0) nums1[l --] = nums1[m --];
        while(n >= 0) nums1[l --] = nums2[n --];
    }
}